diff --git a/cmake/DownloadCEF.cmake b/cmake/DownloadCEF.cmake
index f702268..cbb3a65 100644
--- a/cmake/DownloadCEF.cmake
+++ b/cmake/DownloadCEF.cmake
@@ -41,7 +41,7 @@ function(DownloadCEF platform version download_dir)
     # Extract the binary distribution.
     message(STATUS "Extracting ${CEF_DOWNLOAD_PATH}...")
     execute_process(
-      COMMAND ${CMAKE_COMMAND} -E tar xzf "${CEF_DOWNLOAD_DIR}/${CEF_DOWNLOAD_FILENAME}"
+      COMMAND tar "xf" "${CEF_DOWNLOAD_DIR}/${CEF_DOWNLOAD_FILENAME}" --use-compress-prog=lbzip2
       WORKING_DIRECTORY ${CEF_DOWNLOAD_DIR}
       )
   endif()
diff --git a/tools/make_distrib.sh b/tools/make_distrib.sh
index cd2fd54..6dee431 100755
--- a/tools/make_distrib.sh
+++ b/tools/make_distrib.sh
@@ -87,6 +87,11 @@ else
       mkdir -p "$DISTRIB_LIB_PATH"
     fi
 
+    # Strip unneeded symbols
+    if [[ $1 == "linux32" || $1 == "linux64" ]]; then
+      strip --strip-all $OUT_BINARY_PATH/*.so
+    fi
+
     cp -f "$OUT_BINARY_PATH"/chrome-sandbox "$DISTRIB_LIB_PATH"
     cp -f "$OUT_BINARY_PATH"/libjcef.so "$DISTRIB_LIB_PATH"
     cp -f "$OUT_BINARY_PATH"/jcef_helper "$DISTRIB_LIB_PATH"
diff --git a/tools/make_docs.sh b/tools/make_docs.sh
index 01e4d52..f61aac3 100755
--- a/tools/make_docs.sh
+++ b/tools/make_docs.sh
@@ -6,5 +6,10 @@
 DIR="$( cd "$( dirname "$0" )" && cd .. && pwd )"
 OUT_PATH="${DIR}/out/docs"
 
-javadoc -Xdoclint:none -windowtitle "CEF3 Java API Docs" -footer "<center><a href="https://bitbucket.org/chromiumembedded/java-cef" target="_top">Chromium Embedded Framework (CEF)</a> Copyright &copy 2013 Marshall A. Greenblatt</center>" -nodeprecated -d "$OUT_PATH" -sourcepath "${DIR}/java" -link http://docs.oracle.com/javase/7/docs/api/ -subpackages org.cef
+export CLASSPATH="${DIR}/third_party/jogamp/jar/*"
 
+javadoc -Xdoclint:none -windowtitle "CEF3 Java API Docs" \
+-footer "<center><a href="https://bitbucket.org/chromiumembedded/java-cef" target="_top">Chromium Embedded Framework (CEF)</a> Copyright &copy $(date '+%Y') Marshall A. Greenblatt</center>" \
+-nodeprecated -d "$OUT_PATH" -sourcepath "${DIR}/java" -link 'http://docs.oracle.com/en/java/javase/11/docs/api/' \
+-subpackages org.cef \
+-javafx --ignore-source-errors
diff --git a/tools/make_jar.sh b/tools/make_jar.sh
index 1bc2618..d55d3f1 100755
--- a/tools/make_jar.sh
+++ b/tools/make_jar.sh
@@ -8,6 +8,7 @@ if [ -z "$1" ]; then
 else
   DIR="$( cd "$( dirname "$0" )" && cd .. && pwd )"
   OUT_DIR="${DIR}/out/$1"
+  jar -cf "${OUT_DIR}/jcef-sources.jar" -C "${DIR}/java" org/cef
   jar -cf "${OUT_DIR}"/jcef.jar -C "${OUT_DIR}" org
   jar -cf "${OUT_DIR}"/jcef-tests.jar -C "${OUT_DIR}" tests
 fi
